src/core/reference/src/runtime/reference/generate_proposal.cpp

```diff
@@ -0,0 +1,417 @@
+            int64_t p_idx = (h * bottom_W + w) * anchors_num * 6;
+            int64_t sc_idx = h * bottom_W + w;
+            int64_t d_idx = h * bottom_W + w;
+
+            for (int64_t anchor = 0; anchor < anchors_num; ++anchor) {
+                float x0 = anchors[a_idx + 0];
+                float y0 = anchors[a_idx + 1];
+                float x1 = anchors[a_idx + 2];
+                float y1 = anchors[a_idx + 3];
+
+                const float dx = deltas[d_idx + 0 * bottom_area];
+                const float dy = deltas[d_idx + 1 * bottom_area];
+                const float d_log_w = deltas[d_idx + 2 * bottom_area];
+                const float d_log_h = deltas[d_idx + 3 * bottom_area];
+
+                const float score = scores[sc_idx];
+
+                // width & height of box
+                const float ww = x1 - x0 + coordinates_offset;
+                const float hh = y1 - y0 + coordinates_offset;
+                // center location of box
+                const float ctr_x = x0 + 0.5f * ww;
+                const float ctr_y = y0 + 0.5f * hh;
+
+                // new center location according to deltas (dx, dy)
+                const float pred_ctr_x = dx * ww + ctr_x;
+                const float pred_ctr_y = dy * hh + ctr_y;
+                // new width & height according to deltas d(log w), d(log h)
+                const float pred_w = std::exp(std::min(d_log_w, max_delta_log_wh)) * ww;
+                const float pred_h = std::exp(std::min(d_log_h, max_delta_log_wh)) * hh;
+
+                // update upper-left corner location
+                x0 = pred_ctr_x - 0.5f * pred_w;
+                y0 = pred_ctr_y - 0.5f * pred_h;
+                // update lower-right corner location
+                x1 = pred_ctr_x + 0.5f * pred_w - coordinates_offset;
+                y1 = pred_ctr_y + 0.5f * pred_h - coordinates_offset;
+
+                // adjust new corner locations to be within the image region,
+                x0 = std::max<float>(0.0f, std::min<float>(x0, img_W - coordinates_offset));
+                y0 = std::max<float>(0.0f, std::min<float>(y0, img_H - coordinates_offset));
+                x1 = std::max<float>(0.0f, std::min<float>(x1, img_W - coordinates_offset));
+                y1 = std::max<float>(0.0f, std::min<float>(y1, img_H - coordinates_offset));
+
+                // recompute new width & height
+                const float box_w = x1 - x0 + coordinates_offset;
+                const float box_h = y1 - y0 + coordinates_offset;
+
+                proposals[p_idx + 0] = x0;
+                proposals[p_idx + 1] = y0;
+                proposals[p_idx + 2] = x1;
+                proposals[p_idx + 3] = y1;
+                proposals[p_idx + 4] = score;
+                proposals[p_idx + 5] = (min_box_W <= box_w) * (min_box_H <= box_h) * 1.0;
+
+                a_idx += 4;
+                p_idx += 6;
+                sc_idx += bottom_area;
+                d_idx += 4 * bottom_area;
```

Comment:
Consider adding some comments or names for these magic numbers;)

Reply:
done